File: app/src/main/java/com/android/app/android_baraka_demo/data/network/RetrofitClient.java
package com.android.app.android_baraka_demo.data.network;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static RetrofitClient instance = null;
    private final ApiService myApiService;

    private RetrofitClient() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://www.demo.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        myApiService = retrofit.create(ApiService.class);
    }

    public static synchronized RetrofitClient getInstance() {
        if (instance == null) {
            instance = new RetrofitClient();
        }
        return instance;
    }

    public ApiService getService() {
        return myApiService;
    }
}

File: app/src/main/java/com/android/app/android_baraka_demo/data/network/RetrofitClient.kt
package com.android.app.android_baraka_demo.data.network

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    fun getApiService(): ApiService {
        val retrofit = Retrofit.Builder()
            .baseUrl("https://www.demo.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        return retrofit.create(ApiService::class.java)
    }
}